src/irr/asset/CImageWriterOpenEXR.cpp

```diff
@@ -77,44 +77,30 @@ namespace irr
 				if (region->imageSubresource.mipLevel == 0)
 				{
 					regionsToHandle.push_back(region);
-					width += region->bufferRowLength;
-					height += region->bufferImageHeight;
+					width += region->imageExtent.width;
+					height += region->imageExtent.height;
 				}
 
-			// whole end image that will consists of all pixels gathered in certain regions
-			for (auto channelPixelsPtr : pixelsArrayIlm)
+			for (auto& channelPixelsPtr : pixelsArrayIlm)
 				channelPixelsPtr = _IRR_NEW_ARRAY(ilmType, width * height);
 
-			/// array of single region pixels that will be copied in appropriate way to pixelsArrayIlm to create final image
-			std::vector<std::array<ilmType*, availableChannels>> regionsPixelArray(regionsToHandle.size());
-			for(auto region : regionsToHandle)
-				for (auto regionPixelArray : regionsPixelArray)
-					for (auto channelPixelArray : regionPixelArray)
-						channelPixelArray = _IRR_NEW_ARRAY(ilmType, region->bufferRowLength * region->bufferImageHeight);
-
-			////// TO CHANGE - PUT PIXELS FROM REGIONS TO pixelsArrayIlm
-
+			auto data = image->getBuffer()->getPointer();
 			for (auto region : regionsToHandle)
 			{
-				for (uint64_t yPos = 0; yPos < region->bufferImageHeight; ++yPos)
-					for (uint64_t xPos = 0; xPos < region->bufferRowLength; ++xPos)
+				for (uint64_t yPos = region->imageOffset.y; yPos < region->imageOffset.y + region->imageExtent.height; ++yPos)
+					for (uint64_t xPos = region->imageOffset.x; xPos < region->imageOffset.x + region->imageExtent.width; ++xPos)
 					{
-						/*
-							TODO - compute positions
-						
-						const uint64_t ptrStyleEndShiftToImageDataPixel = region->bufferOffset + (yPos * region->bufferRowLength * availableChannels) + (xPos * availableChannels);
+						const uint64_t ptrStyleEndShiftToImageDataPixel = (yPos * width * availableChannels) + (xPos * availableChannels);
 						const uint64_t ptrStyleIlmShiftToDataChannelPixel = (yPos * width) + xPos;
 
 						for (uint8_t channelIndex = 0; channelIndex < availableChannels; ++channelIndex)
 						{
-							ilmType channelPixel = *(reinterpret_cast<const ilmType*>(image->getBuffer()->getPointer()) + ptrStyleEndShiftToImageDataPixel + channelIndex);
+							ilmType channelPixel = *(reinterpret_cast<const ilmType*>(data) + ptrStyleEndShiftToImageDataPixel + channelIndex);
```

Comment:
region also specifies a byte offset into the beginning of the buffer.

Reply:
I know, but isn't it good? If not, how should I combine a byte offset into the beginning of the buffer with `ptrStyleEndShiftToImageDataPixel`?

Reply:
Do I need it if I specify a begginging of each pixel using `ptrStyleEndShiftToImageDataPixel`?